frontend/js/src/common/brainzplayer/BrainzPlayer.tsx

```diff
@@ -945,6 +951,75 @@ export default function BrainzPlayer() {
     }
   };
 
+  const handleAlbumMapping = (
+    dataSource: keyof MatchedTrack,
+    releaseMBID: string,
+    album: {
+      trackName: string;
+      uri: string;
+    }[]
+  ): void => {
+    const currentQueue = queueRef.current;
+    const currentAmbientQueue = ambientQueueRef.current;
+
+    // Filter the currentQueue after the current playing track
+    const currentPlayingListenIndex = currentListenIndexRef.current;
+    const currentQueueAfterCurrent = currentQueue.slice(
```

Comment:
I reckon this needs a more descriptive name.
Maybe something like `upcomingQueue` or `upcomingQueueItems`?